src/json_schema/validators/of.rs

```diff
@@ -12,17 +12,56 @@ pub struct AllOf {
 impl super::Validator for AllOf {
     fn validate(&self, val: &Value, path: &str, scope: &scope::Scope) -> super::ValidationState {
         let mut state = super::ValidationState::new();
+        let mut val = Cow::Borrowed(val);
+
+        // first get all relevant schemas
+        let schemas = self
+            .schemes
+            .iter()
+            .map(|url| (url, scope.resolve(url)))
+            .filter_map(|(url, opt)| {
+                if opt.is_none() {
+                    state.missing.push(url.clone())
+                }
+                opt
+            })
+            .collect::<Vec<_>>();
+
+        // first pass to populate all defaults (if enabled)
+        for schema in schemas.iter() {
+            let mut result = schema.validate_in(&val, path);
+            if result.is_valid() && result.replacement.is_some() {
+                *val.to_mut() = result.replacement.take().unwrap();
+            }
+            state.append(result);
+        }
+        if !state.is_valid() {
+            return state;
+        }
 
-        for url in self.schemes.iter() {
-            let schema = scope.resolve(url);
+        // second pass if defaults are enabled to check that the result is stable
```

Comment:
What is an example of a schema where you might have divergent defaults? And can we add it as a test case?

Reply:
good point, I shall try to make a small one